devito/petsc/types.py

```diff
@@ -65,6 +94,12 @@ class SNES(LocalObject):
     """
     dtype = CustomDtype('SNES')
 
+    @property
+    def _C_free(self):
+        from devito.petsc.utils import petsc_call
```

Comment:
since you're in control of what happens within the `petsc/` module, this nested import shouldn't be to challenging to avoid?